### PR TITLE
fix calc property removing

### PR DESCRIPTION
**Pull request by ArturAralin**

### CODE REVIEW

File: lib/compressor/restructure/4-restructShorthand.js [hunk 1]
```diff
@@ -365,6 +368,7 @@ function processShorthands(shortDeclarations, markDeclaration) {
         }
 
         if (item.operation === REPLACE) {
+
```
Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Don't add/remove blank line that's not related to your changes

Comment (ArturAralin) [inline lib/compressor/restructure/4-restructShorthand.js]: sry

File: lib/compressor/restructure/4-restructShorthand.js [hunk 2]
```diff
@@ -113,6 +113,9 @@ TRBL.prototype.getValueSequence = function(value, count) {
                         break;
                 }
                 break;
+            case 'Function':
```
Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Add blank line before `case` please

**Comment by lahmatiy** [comment]
Add a test for fixed case, please

File: lib/compressor/restructure/4-restructShorthand.js [hunk 3]
```diff
@@ -121,6 +121,12 @@ TRBL.prototype.getValueSequence = function(value, count) {
             case 'Space':
                 return false; // ignore space
 
+            case 'Function':
+                switch (child.name) {
```
Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Why switch is here? It isn't necessary

File: lib/compressor/restructure/4-restructShorthand.js [hunk 4]
```diff
@@ -121,6 +121,12 @@ TRBL.prototype.getValueSequence = function(value, count) {
...
+                    case 'calc':
+                        special = child.name;
+                        break;
+                }
```
Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Missed `break` and blank line before `default`

Comment (ArturAralin) [inline lib/compressor/restructure/4-restructShorthand.js]: where i miss break operation? and style tests says that's code style is valid

Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]:
```js
            case 'Number':
            case 'Percentage':
                break;   // <- should be blank line before `case`, look on other switch branches
            case 'Function':
                switch (child.name) {
                    case 'calc':
                        special = child.name;
                        break;
                }
                // <- here is should be break
            case 'Space':
                return false; // ignore space
```

Some things aren't covered by style checks. Look on other code is formatted in this case.

Comment (ArturAralin) [inline lib/compressor/restructure/4-restructShorthand.js]: If i add brake operation then tests will be failed

**Comment by ArturAralin** [comment]
wtf? Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}

File: test/compress.js [hunk 5]
```diff
@@ -122,6 +122,15 @@ describe('compress', function() {
         });
     });
 
+    describe('calc function', function() {
```
Comment (lahmatiy) [inline test/compress.js]: Please, read my comment https://github.com/css/csso/pull/303#issuecomment-226820144

File: lib/compressor/restructure/4-restructShorthand.js [hunk 6]
```diff
@@ -118,6 +118,16 @@ TRBL.prototype.getValueSequence = function(value, count) {
             case 'Percentage':
                 break;
 
+            case 'Function':
+                switch (child.name) {
```
Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Why switch is here?

```js
if (child.name !== 'calc') {
    return false;
}

special = child.name;
break;
```

Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: Also looks like 

```js
special = child.name;
break;
```

is enough and improve optimisations for values with function

Comment (ArturAralin) [inline lib/compressor/restructure/4-restructShorthand.js]: may appear new rules to other function therefore i added switch construction without if construction

Comment (lahmatiy) [inline lib/compressor/restructure/4-restructShorthand.js]: > may appear new rules to other functions

`switch` may to be added than – not now

File: test/compress.js (hunk 5 above)
Comment (ArturAralin) [inline test/compress.js]: I missed it

File: lib/compressor/restructure/4-restructShorthand.js (hunk 6 above)
Comment (ArturAralin) [inline lib/compressor/restructure/4-restructShorthand.js]: i don't see significant difference

File: test/compress.js (hunk 5 above)
Comment (lahmatiy) [inline test/compress.js]: This code doesn't needed anymore

Comment (ArturAralin) [inline test/compress.js]: > This code doesn't needed anymore

ok

**Pull request merged**